### PR TITLE
Added more hotkeys to the editor

### LINKED ISSUE #10135: Hotkeys to swap selected tile in editor
As an example: up and down arrows to change category, left and right to move between tiles. This would GREATLY speed up mapping.

This could of course be further expanded with other hoykeys, but this seems like the first and most important one to me as it is very slow to manually find your tile every time.

Other potential hotkeys:
* Copy and paste.
* Tile/Overlay/Actors menu swapping.
* Save map.
* Zoom.

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/10135

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/ScrollPanelWidget.cs [hunk 1]
```diff
@@ -392,6 +396,58 @@ public override bool HandleMouseInput(MouseInput mi)
...
+			if (ScrollDownKey.IsActivatedBy(e))
+			{
+				Scroll(-1);
+				return true;
+			}
+
+			var selectedItem = (ScrollItemWidget)Children.FirstOrDefault(c =>
+			{
+				return c is ScrollItemWidget si && si.IsSelected();
+			});
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/ScrollPanelWidget.cs]:
```suggestion
			var selectedItem = (ScrollItemWidget)Children.FirstOrDefault(c => c is ScrollItemWidget si && si.IsSelected());
```
?

Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/ScrollPanelWidget.cs]: Yes, thanks.

**Review by dragunoff — COMMENTED** [review]
This is a great UX improvement for the editor but I think it should be implemented in a different way :thinking: Here are some thoughts I have after testing and looking at the code:
- I think keyboard navigation should be handled inside widgets only when the widget has focus. Thus scroll panel navigation should be solved as a part of a holistic system for keyboard navigation. And then the <kbd>Tab</kbd> key probably needs to be reserved for switching focus.
- Scroll panels can still be activated by hotkeys but handled in a chrome logic class. For example a logic class that has hotkeys for cycling the tiles/actors of the scroll panel to which the current brush is set to (even if the scroll panel is not visible).
- Scrolling up/down doesn't seem that useful to me so IMO we should just focus on the cycling part.
- It would be great if the hotkeys in the editor are added as named hotkeys and listed in the hotkey settigs.

**Comment by PunkPun** [comment]
Can PR consider this PR superseded?

**Comment by dragunoff** [comment]
> Can PR consider this PR superseded?

Which PR do you have in mind? I don't know of any PR that supersedes this one.

**Comment by PunkPun** [comment]
Pretty much every feature that is being added in this PR, is already either on release or bleed

**Comment by dragunoff** [comment]
> Pretty much every feature that is being added in this PR, is already either on release or bleed

This PR adds a quick save button and hotkeys to scroll the active pane and to cycle items in it (tiles, actors, etc.). I'm not sure if quick save is already on bleed but the hotkeys for items cycling certainly isn't.

**Comment by PunkPun** [comment]
hmm, ok

**Comment by dragunoff** [comment]
@Mailaender I suggest you split the quick save button to its own PR. It's a simple feature that has a much higher chance of getting merged separately.

**Comment by Mailaender** [comment]
There you go https://github.com/OpenRA/OpenRA/pull/20204

**Comment by PunkPun** [comment]
needs a rebase

**Comment by PunkPun** [comment]
> It would be great if the hotkeys in the editor are added as named hotkeys and listed in the hotkey settigs.

I agree. These hotkeys should definitely be made rebindable. I can't even test out all the features with defaults without hooking up an external keyboard 😅

**Pull request closed without merging**